### PR TITLE
Add PHPStorm Inspections as a CI build step

### PR DESCRIPTION
**Pull request by mridang**
## Checklist:
- [ ] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [ ] I have assigned the correct milestone or created one if non-existent.
- [ ] I have correctly labeled this pull request.
- [ ] I have linked the corresponding issue in this description.
- [ ] I have updated the corresponding Jira ticket.
- [ ] I have requested a review from at least 2 reviewers
- [ ] I have checked the base branch of this pull request
- [ ] I have checked my code for any possible security vulnerabilities

### CODE REVIEW

File: classes/blocks/NostoPageTypeTagging.php [hunk 1]
```diff
@@ -23,6 +23,7 @@
  * @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
  */
 
+use Nosto\NostoException;
```
Comment (supercid) [inline classes/blocks/NostoPageTypeTagging.php:26]: This looks incorrect

File: classes/helpers/NostoHelperContext.php [hunk 2]
```diff
@@ -57,7 +57,7 @@ public static function runInContext(
         try {
             // @phan-suppress-next-line PhanTypeVoidAssignment
             $retVal = $callable();
-        } catch (\Exception $e) {
+        } catch (Exception $e) {
```
Comment (supercid) [inline classes/helpers/NostoHelperContext.php:60]: Looks incorrect

File: classes/helpers/NostoHelperController.php [hunk 3]
```diff
@@ -80,7 +80,7 @@ public static function resolveObject($idName, $klass, $method, $multiLanguageObj
         if ($object instanceof $klass === false && $id !== null) {
             try {
                 $reflectionMethod = new ReflectionMethod($klass, '__construct');
-            } catch (\ReflectionException $e) {
+            } catch (ReflectionException $e) {
```
Comment (supercid) [inline classes/helpers/NostoHelperController.php:83]: Looks incorrect

**Comment by supercid** [comment]
image

File: classes/helpers/NostoHelperCurrency.php [hunk 4]
```diff
@@ -230,6 +242,40 @@ private static function createWithCldr(array $currency)
...
+            throw new NostoException(
+                sprintf(
+                    'Could not find default %s symbol to use in currency formatting',
+                    self::DEFAULT_SYMBOL_FIELD
+                )
+            );
+        }
+        $numberSymbolList = $symbols[self::DEFAULT_SYMBOL_FIELD];
+        $symbolPos = Tools::strpos($priceSpec->getPositivePattern(), self::CURRENCY_SYMBOL_MARKER);
+        $symbolPosition = $symbolPos === 0;
```
Comment (supercid) [inline classes/helpers/NostoHelperCurrency.php:266]: I'd rename this var to clarify, like
```suggestion
        $isCurrencyBeforeAmount = $symbolPos === 0;
```

**Review by supercid — APPROVED** [review]

File: classes/helpers/NostoHelperCurrency.php (hunk 4 above)
Comment (hannunosto) [reply classes/helpers/NostoHelperCurrency.php]: Dude, thanks for the input, makes sense :)

**Review by hannunosto — APPROVED** [review]

File: controllers/front/api.php [hunk 5]
```diff
@@ -65,6 +66,7 @@ public function __construct()
      *
      * @param Nosto\Model\AbstractCollection $collection the data collection to output as
      *     encrypted response.
+     * @throws NostoException
      */
     public function encryptOutput(Nosto\Model\AbstractCollection $collection)
```
Comment (hannunosto) [inline controllers/front/api.php:71]: You could simplify FQN

File: ctrl.php [hunk 6]
```diff
@@ -74,7 +74,15 @@
                 die();
         }
 
-        $nostoController->initContent();
+        try {
+            $nostoController->initContent();
+        } catch (Exception $e) {
+            die(sprintf(
```
Comment (hannunosto) [inline ctrl.php:80]: Add a new line here

File: views/js/nostotagging-admin-config.js [hunk 7]
```diff
@@ -22,23 +22,31 @@
...
+// noinspection JSUnresolvedFunction
 $(document).ready(function () {
     // Change event handler for "Manage Accounts:".
+    // noinspection JSUnresolvedFunction
     $("#nostotagging_language").change(function () {
-        var langId = parseInt($(this).val()),
-            $currentLanguage = $('#nostotagging_current_language'),
-            $form = $('form.nostotagging');
+      // noinspection JSUnresolvedFunction
+        const langId = parseInt($(this).val()),
```
Comment (hannunosto) [inline views/js/nostotagging-admin-config.js:31]: Are we sure that we only want to support ES 6 compatible browsers?

Comment (mridang) [reply views/js/nostotagging-admin-config.js:31]: We can change that to `var`.

Comment (hannunosto) [reply views/js/nostotagging-admin-config.js:31]: 👍

**Review by hannunosto — APPROVED** [review]
Looks good to me 👍

File: composer.json [hunk 8]
```diff
@@ -5,7 +5,12 @@
     "AFL-3.0"
   ],
   "require": {
-    "nosto/php-sdk": "5.1.*"
+    "nosto/php-sdk": "5.1.*",
+    "ext-json": "*",
```
Comment (mridang) [inline composer.json:9]: @hannunosto This might cause issues

File: classes/helpers/NostoHelperCurrency.php [hunk 9]
```diff
@@ -1,4 +1,7 @@
-<?php
+<?php /** @noinspection PhpUnused */
+/** @noinspection PhpUnused */
```
Comment (olsi-qose) [inline classes/helpers/NostoHelperCurrency.php:2]: Repeated 3 times

File: composer.json (hunk 8 above)
Comment (hannunosto) [reply composer.json:9]: This should not cause any problems. Composer is only used in packaging and in module installation.

File: views/templates/hook/header_prestashop-add-to-cart-event-handler.tpl [hunk 10]
```diff
@@ -22,14 +22,16 @@
 * @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
 *}
 
+<!--suppress JSUnresolvedFunction, JSUnresolvedVariable -->
 <script type="text/javascript">
-    nostojs(function(api){
```
Comment (olsi-qose) [inline views/templates/hook/header_prestashop-add-to-cart-event-handler.tpl:26]: Why is the api param removed?

**Review by olsi-qose — APPROVED** [review]

**Pull request merged**